tweetapi/views.py

```diff
@@ -20,37 +19,31 @@ class ProfileView(generics.ListCreateAPIView):
     serializer_class = ProfileSerilizer
 
 class ProfileDetails(generics.RetrieveUpdateDestroyAPIView):
-    authentication_classes = [Authentication]
     permission_classes = [TwitterUserPermission]
     queryset = ProfileModel.objects.all()
     serializer_class = ProfileSerilizer
 
 class TweetView(generics.ListCreateAPIView):
-    authentication_classes = [Authentication]
+    permission_classes = [AllowAny]
     queryset = TweetModel.objects.all()
     serializer_class = TweetSerializer
 
 class TweetViewDetails(generics.RetrieveDestroyAPIView):
-    permission_classes = [TwitterUserPermission]
-    authentication_classes = [Authentication]
     queryset = TweetModel.objects.all()
     serializer_class= TweetSerializer
 
 class CommentView(generics.ListCreateAPIView):
     permission_classes = [TwitterUserPermission]
-    authentication_classes = [Authentication]
     queryset = CommentModel.objects.all()
     serializer_class = CommentSerializer
 
 class CommentDetails(generics.RetrieveDestroyAPIView):
     permission_classes = [TwitterUserPermission]
-    authentication_classes = [Authentication]
     queryset = CommentModel.objects.all()
     serializer_class = CommentSerializer
 
 class LikeView(generics.ListCreateAPIView):
     permission_classes = [TwitterUserPermission]
```

Comment:
when you set permission class like this does the default one in setting still apply?

Reply:
yes sir it does still apply, because if the owner is not authenticated they cant perform anything.

Reply:
Are you sure about this. make sure to check